lib/order.rb

```diff
@@ -0,0 +1,22 @@
+class Order
+
+  def initialize
+    @choosen_dishes = []
+  end
+
+  def add(dish_quantity)
+    choosen_dishes << dish_quantity
+    # @choosen_dishes << {name: dish[:name],price: dish[:price], quantity: quantity}
+  end
+
+  def calculate_total
+    choosen_dishes.map{|info| info[:price]*info[:quantity]}.reduce(:+)
```

Comment:
Really powerful use of map and reduce.  To make it easier to read, try breaking it into multiple lines:

```
choosen_dishes
  .map{|info| info[:price] * info[:quantity]}
  .reduce(:+)
```